runtime/src/lib.rs

```diff
@@ -376,6 +189,18 @@ pub fn handle_mailbox_commands(drivers: &mut Drivers) -> CaliptraResult<()> {
     }
     loop {
         enter_idle(drivers);
+        if drivers.is_shutdown {
+            cprintln!("[rt] is_shutdown");
+            caliptra_drivers::report_fw_error_non_fatal(CaliptraError::RUNTIME_SHUTDOWN.into());
+            caliptra_drivers::report_fw_error_fatal(CaliptraError::RUNTIME_SHUTDOWN.into());
+            loop {
+                // SoC firmware might be stuck waiting for Caliptra to finish
+                // executing this pending mailbox transaction. Notify them that
+                // we've failed.
+                unsafe { caliptra_drivers::Mailbox::abort_pending_soc_to_uc_transactions() };
+            }
```

Comment:
Instead of doing this, we should be able to just fail this command and `continue`. On the next mailbox command we'll hit this if condition again and fail that command.

Reply:
The semantics of receiving a command in shutdown should be handled the same way as ROM. In ROM we are handling as error.

Reply:
Right, I'm not suggesting changing that behavior. I just think the code could be written in a way that's more clear what's happening. Something like:

```rust
loop {
    enter_idle(drivers);
    if drivers.mbox.is_cmd_ready() {
        if drivers.is_shutdown {
            cprintln!("[rt] is_shutdown");
            caliptra_drivers::report_fw_error_non_fatal(CaliptraError::RUNTIME_SHUTDOWN.into());
            caliptra_drivers::report_fw_error_fatal(CaliptraError::RUNTIME_SHUTDOWN.into());
            unsafe { caliptra_drivers::Mailbox::abort_pending_soc_to_uc_transactions() };
            continue;
        }
        ...
    }
}
```

This will ensure every mailbox command is failed, but avoids having another infinite loop. I will also make managing the watchdog easier (especially when it is moved to `enter_idle`) since the normal watchdog behavior will still work.